backend/fms_core/tests/test_template_importers/sample_pooling_test.py

```diff
@@ -0,0 +1,191 @@
+from django.test import TestCase
+from datetime import datetime
+from decimal import Decimal
+
+from fms_core.template_importer.importers import SamplePoolingImporter
+from fms_core.tests.test_template_importers._utils import load_template, APP_DATA_ROOT
+from fms_core.tests.constants import create_individual
+
+from fms_core.models import SampleKind, Protocol, Process, ProcessMeasurement, Individual, Sample, SampleLineage, DerivedSample, DerivedBySample
+from fms_core.models._constants import DOUBLE_STRANDED
+
+from fms_core.services.container import create_container
+from fms_core.services.index import get_or_create_index_set, create_index
+from fms_core.services.library import create_library, get_library_type
+from fms_core.services.platform import get_platform
+from fms_core.services.sample import create_full_sample
+
+class SamplePoolingTestCase(TestCase):
+    def setUp(self) -> None:
+        self.importer = SamplePoolingImporter()
+        self.file = APP_DATA_ROOT / "Sample_pooling_v3_12_0.xlsx"
+
+        self.DNA_sample_kind, _ = SampleKind.objects.get_or_create(name='DNA')
+        self.RNA_sample_kind, _ = SampleKind.objects.get_or_create(name="RNA")
+        self.protocol_pooling = Protocol.objects.get(name="Sample Pooling")
+
+        self.plate_source_name_and_barcode = "POOLFROMPLATE1"
+        self.source_sample_initial_volume = 200
+
+        self.prefill_data()
+
+
+    def prefill_data(self):
+
+        container, _, _ = create_container(barcode=self.plate_source_name_and_barcode,
+                                           kind='96-well plate',
+                                           name=self.plate_source_name_and_barcode)
+
+        # index
+        index_set, _, _, _ = get_or_create_index_set("PoolTestSet")
+        self.index_1, _, _ = create_index(index_name="PoolLibIndex1", index_structure="TruSeqLT", index_set=index_set)
+        self.index_2, _, _ = create_index(index_name="PoolLibIndex2", index_structure="TruSeqLT", index_set=index_set)
+        self.index_3, _, _ = create_index(index_name="PoolLibIndex3", index_structure="TruSeqLT", index_set=index_set)
+
+        # library type
+        library_type, _, _ = get_library_type(name="PCR-free")
+
+        # platform
+        platform, _, _ = get_platform(name="ILLUMINA")
+
+
+        library_1, _, _ = create_library(index=self.index_1, library_type=library_type, platform=platform, strandedness=DOUBLE_STRANDED, library_size=150)
+        library_2, _, _ = create_library(index=self.index_2, library_type=library_type, platform=platform, strandedness=DOUBLE_STRANDED, library_size=150)
+        library_3, _, _ = create_library(index=self.index_3, library_type=library_type, platform=platform, strandedness=DOUBLE_STRANDED, library_size=150)
+
+        self.source_sample_1, _, _ = \
+            create_full_sample(name="SOURCESAMPLE1POOL1", alias="SOURCESAMPLE1POOL1", volume=self.source_sample_initial_volume, concentration=25,
```

Comment:
It might be good to have some non-integer values in our tests (eg. 25.333). I see linter complaints about illegal decimal/integer conversions, but I assume that the linter is just confused. Still, it would be good to test that decimal values are not being accidentally rounded down to integer values in our code.

Reply:
Trouble is that comparing floats and decimal is not very safe (25.333 may not be equal Decimal('25.333'). If we use fractional values, we should use Decimal instead which in turn defeats the purpose.